tests/callbacks/test_optimizer_monitor.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+from torch.utils.data import DataLoader
+
+from composer.callbacks import OptimizerMonitor
+from composer.loggers import InMemoryLogger
+from composer.optim import DecoupledAdamW
+from composer.trainer import Trainer
+from tests.common.datasets import RandomClassificationDataset
+from tests.common.models import SimpleModel
+
+
+@pytest.mark.parametrize('log_optimizer_metrics', [True, False])
+def test_optimizer_monitor(log_optimizer_metrics: bool):
+    # Construct the callback
+    grad_monitor = OptimizerMonitor(log_optimizer_metrics=log_optimizer_metrics)
+    in_memory_logger = InMemoryLogger()  # track the logged metrics in the in_memory_logger
+    model = SimpleModel()
+    # Construct the trainer and train
+    trainer = Trainer(
+        model=model,
+        callbacks=grad_monitor,
+        loggers=in_memory_logger,
+        train_dataloader=DataLoader(RandomClassificationDataset()),
+        optimizers=DecoupledAdamW(model.parameters()),
+        max_duration='1ep',
```

Comment:
nit: since this logs per batch, can we instead set max duration to `3ba` or something? 

We're cleaning up our slow tests to try to speedup CI/CD workflow, and smaller fit runs are always helpful

Reply:
good suggestions!